agent/engine/task_manager.go

```diff
@@ -246,24 +246,32 @@ func (mtask *managedTask) handleContainerChange(containerChange dockerContainerC
 			container.ApplyingError = api.NewNamedError(event.Error)
 		}
 		if event.Status == api.ContainerStopped {
-			// If we were trying to transition to stopped and had a timeout error
-			// from docker, reset the known status to the current status and return
+			eventErrorName := event.Error.ErrorName()
+			// If we were trying to transition to stopped and had a transient error
+			// from docker (example: timeout), reset the known status to the current
+			// status and return.
+			// Errors such as "No such container", "Container already stopped" etc
+			// from docker do not fall into this category. Nor do errors such as
+			// "CannotStartContainer" or "DockerStateError" ("invalid executable
+			// in path").
 			// This ensures that we don't emit a containerstopped event; a
 			// terminal container event from docker event stream will instead be
 			// responsible for the transition. Alternatively, the steadyState check
 			// could also trigger the progress and have another go at stopping the
 			// container
-			if event.Error.ErrorName() == dockerTimeoutErrorName {
-				seelog.Infof("%s for 'docker stop' of container; ignoring state change;  task: %v, container: %v, error: %v", dockerTimeoutErrorName, mtask.Task, container, event.Error.Error())
+			if eventErrorName == "CannotStopContainerError" {
```

Comment:
+1

Also, can you do type assertions instead of string comparison?  You could assert to `CannotXContainerError` and then string match "Stop" if necessary.